qiskit/providers/fake_provider/fake_pulse_backend.py

```diff
@@ -26,6 +28,22 @@ class FakePulseBackend(FakeQasmBackend):
 
     defs_filename = None
 
+    def __init__(self):
+        super().__init__()
+        # This is a deprecation warning for the subclasses.
```

Comment:
Is this class still used by anything in Qiskit? And, how about `FakeBackendV1` which this ultimately uses as a base class? Can all of this be deprecated?

Reply:
Ah I see, you're doing this in a separate PR it looks like 😄

Reply:
This is confusing, but the V1 base classes (`FakeBackendV1`, `FakePulseBackend`, `FakeQasmBackend`) cannot be deprecated yet because they are used in the "generic" backend version too. The "fake backend alternatives" in #10952 are effectively snapshot-based fake backends with "more generic" names. This was agreed to be the best temporary alternative while BackendV1 is around, the plan is to eventually fully transition to BackendV2 but this will not happen for 1.0.

So now we have the challenge to deprecate ~45 subclasses without deprecating the base classes. This warning is only for the subclasses, and this is why I don't use the decorator here, so that the message doesn't say `"FakeBackendV1 is deprecated"` but `"All fake backend instances based on real device snapshots..."`. And about the V1 hierarchy, I am taking advantage of the fact that these "V1 alternatives" to fake backends live on the main branch and the deprecations live on 0.46, so the warning would never appear on the non-deprecated alternatives.

Reply:
Ah okay, thank you! I was missing that subtly 😄.